base/linalg/tridiag.jl

```diff
@@ -312,7 +321,19 @@ immutable Tridiagonal{T} <: AbstractMatrix{T}
     d::Vector{T}     # diagonal
     du::Vector{T}    # sup-diagonal
     du2::Vector{T}   # supsup-diagonal for pivoting
+
+    Tridiagonal(dl::Vector{T}, d::Vector{T}, du::Vector{T}, du2::Vector{T}) = new(dl, d, du, du2)
+    function Tridiagonal(n::Integer, m::Integer)
+        if n != m
+            throw(DimensionMismatch("Tridiagonal matrix must be square"))
+        end
+        new(Vector{T}(n-1), Vector{T}(n), Vector{T}(n-1), zeros(T, n-2))
```

Comment:
Thanks for confirming; I've removed the initialization, will be part of next push.